src/listen/websocket.rs

```diff
@@ -238,12 +257,60 @@ fn options_to_query_string(options: &Options) -> String {
     serde_urlencoded::to_string(serialized_options).unwrap_or_default()
 }
 
+#[derive(Debug)]
+pub struct TranscriptionStream {
+    write_arc: Arc<Mutex<SplitSink<WebSocketStream<MaybeTlsStream<TcpStream>>, Message>>>,
+}
+
+impl TranscriptionStream {
+    pub async fn finalize(
+        &self,
+        event_tx: Sender<Event>,
+    ) -> std::result::Result<(), DeepgramError> {
+        let finalize_message = Message::Text("{\"type\": \"Finalize\"}".to_string());
+        let mut write_guard = self.write_arc.lock().await;
+        if let Err(e) = write_guard.send(finalize_message).await {
+            let err = DeepgramError::from(e);
+            eprintln!("Error sending Finalize message: {:?}", err);
+            event_tx.send(Event::Error(err)).await.unwrap();
+            return Err(DeepgramError::CustomError(
+                "Failed to send Finalize message".to_string(),
+            ));
+        }
+        Ok(())
+    }
+
+    pub async fn finish(&self, event_tx: Sender<Event>) -> std::result::Result<(), DeepgramError> {
```

Comment:
Let's call this `close()` instead of `finish()`.  Finish is too similar to `finalize()` for my tastes, and doesn't match our docs.

Reply:
This matches the other SDKs.

See: https://github.com/deepgram/deepgram-python-sdk/blob/main/examples/speech-to-text/websocket/microphone/main.py#L134

This is not actually a close of the websocket it tells the websocket to finish processing all it's audio and then the server closes